src/pepper/abstract_symbol_tree.py

```diff
@@ -87,78 +71,84 @@ def preprocess(self, lines):
             symtable.FILE_STACK.append(open(os.path.split(symtable.FILE_STACK[-1].name)[0]
                                             + '/' + self.target, 'r'))
 
+        return 'static_assert(false, "include node not properly implemented")'
+
 
 class IdentifierNode(Node):
 
-    def __init__(self, children, args=None, variadic=False):
+    def __init__(self,
+                 children: List[str] = [],
+                 args: Optional[List[Node]] = None,
+                 variadic: bool = False) -> None:
         super(IdentifierNode, self).__init__("Identifier", children)
         self.args = args
         self.variadic = variadic
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"{self.name}: {self.children}"
 
-    def preprocess(self, lines=None):
-        expansion = self.children[0]
+    def preprocess(self, lines: Optional[List[str]] = None) -> str:
+        expansion = ""
         if self.children[0] in symtable.TABLE.keys():
-            expansion = symtable.TABLE[self.children[0]].expand(
+            expansion = symtable.TABLE[cast(str, self.children[0])].expand(
```

Comment:
are children not already strings? why do you need to cast it? is this some way of enforcing the type?

Reply:
I was running into trouble with the output of children.expand, since expand has a bunch of different types. I don't remember specifically why, it's been a while.